com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -386,6 +382,8 @@ private void Init(bool server)
             NetworkConfig.NetworkTransport.ResetChannelCache();
 
             NetworkConfig.NetworkTransport.Init();
+
+            ProfilerNotifier.Initialize(this);
```

Comment:
Is this something we always do even if we're not running in the editor?  Or put another way, what if I'm shipping the game and don't want the profiler overhead?

Reply:
In the same vein, in NetworkProfiler.cs circa line 651 (and I know this is from a previous commit) we unconditionally call `ProfilerBeginTick`

Reply:
So this was an intentional choice on our end, since there may be runtime tools that need this data. That doesn't mean users might not want to omit the profiling from the build, but we haven't provided that utility yet. Since we're just tracking counters in v1 the overhead is very small, but as we expand the data set this is going to be important. Maybe we could add a #if !MLAPI_DISABLE_STATS so that it's on by default, but users can turn it off with the compiler configuration.

I'd suggest that this belongs in a separate PR though. Kamau isn't introducing profiling into release builds with this PR, he's just refactoring functionality that already exists.